rpc-interface/src/types.rs

```diff
@@ -692,6 +693,143 @@ impl Validator {
     }
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BlockchainState<T> {
+    pub block_number: u32,
+    pub block_hash: Blake2bHash,
+    #[serde(flatten)]
+    pub value: T,
+}
+
+impl<T> BlockchainState<T> {
+    pub fn from_value(block_number: u32, block_hash: Blake2bHash, value: T) -> Self {
```

Comment:
Small nitpick.
```suggestion
    pub fn new(block_number: u32, block_hash: Blake2bHash, value: T) -> Self {
```
But since this does not contain any logic and the fields are `pub`, you can also just use 

```rust
BlockchainState { block_number, block_hash, value }
```